icd/api/pipeline_compiler.cpp

```diff
@@ -1047,24 +1047,89 @@ void PipelineCompiler::GetPipelineCreationInfoNext(
 // =====================================================================================================================
 VkResult PipelineCompiler::SetPipelineCreationFeedbackInfo(
     const VkPipelineCreationFeedbackCreateInfoEXT* pPipelineCreationFeadbackCreateInfo,
-    const PipelineCreationFeedback*                pPipelineFeedback)
+    const VkGraphicsPipelineCreateInfo*            pCreateInfo,
+    const PipelineCreationFeedback*                pPipelineFeedback,
+    const PipelineCreationFeedback*                pStageFeedback)
 {
     if (pPipelineCreationFeadbackCreateInfo != nullptr)
     {
+        auto *pipelineCreationFeedback = pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback;
+        pipelineCreationFeedback->flags = 0;
         if (pPipelineFeedback->feedbackValid)
         {
-            pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
-                VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+            pipelineCreationFeedback->flags |= VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
         }
         if (pPipelineFeedback->hitApplicationCache)
         {
-            pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
+            pipelineCreationFeedback->flags |=
                 VK_PIPELINE_CREATION_FEEDBACK_APPLICATION_PIPELINE_CACHE_HIT_BIT_EXT;
         }
-        pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->duration =
-            pPipelineFeedback->duration;
-    }
+        pipelineCreationFeedback->duration = pPipelineFeedback->duration;
+
+        if (pCreateInfo == nullptr) {
+            stageCreationFeedbacks[0].flags = 0;
+        } else {
+            for (uint32_t i = 0; i < pCreateInfo->stageCount; ++i)
+            {
+                stageCreationFeedbacks[i].flags = 0;
+            }
+        }
 
+        if ((pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks != nullptr) &&
+            (pStageFeedback != nullptr))
```

Comment:
pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks can't be null.  I don't see anything in the spec that allows compute pipelines to not have per stage information so we need to handle it.